server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -2791,16 +2792,60 @@ public InputStream maybeRateLimitSnapshots(InputStream stream) {
 
     @Override
     public IndexShardSnapshotStatus getShardSnapshotStatus(SnapshotId snapshotId, IndexId indexId, ShardId shardId) {
-        BlobStoreIndexShardSnapshot snapshot = loadShardSnapshot(shardContainer(indexId, shardId), snapshotId);
-        return IndexShardSnapshotStatus.newDone(
-            snapshot.startTime(),
-            snapshot.time(),
-            snapshot.incrementalFileCount(),
-            snapshot.totalFileCount(),
-            snapshot.incrementalSize(),
-            snapshot.totalSize(),
-            null
-        ); // Not adding a real generation here as it doesn't matter to callers
+        final StepListener<RepositoryData> repoDataListener = new StepListener<>();
+        getRepositoryData(repoDataListener);
+        AtomicBoolean isRemoteIndexShard = new AtomicBoolean(false);
```

Comment:
Why AtomicBoolean, when the context is local?

Reply:
Since we are trying to update the boolean `isRemoteIndexShard` from within the lambda expression in `repoDataListener` listener's whenComplete, I'm using AtomicBoolean to treat it as effectively final.

Reply:
The method getRepositoryData can be async https://github.com/opensearch-project/OpenSearch/blob/a81ef5aa322faf46e7c8be776571612b93c646f5/server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java#L1622, which means the condition 
```
if (!isRemoteIndexShard.get()) {
  blah...
}
```
is not guaranteed to be using the new updated(after the listeners#whenComplete gets invoked) value.